core/src/apps/solana/definitions.py

```diff
@@ -0,0 +1,40 @@
+from typing import TYPE_CHECKING
+
+from trezor.messages import SolanaTokenInfo
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+class Definitions:
+    """Class that holds Solana token definitions."""
+
+    def __init__(self, tokens: dict[bytes, SolanaTokenInfo]) -> None:
+        self._tokens = tokens
+
+    @classmethod
+    def from_encoded(
+        cls,
+        encoded_token: bytes | None,
+    ) -> Self:
+        from apps.common.definitions import decode_definition
+
+        tokens: dict[bytes, SolanaTokenInfo] = {}
+
+        # get token definition
+        if encoded_token is not None:
+            token = decode_definition(encoded_token, SolanaTokenInfo)
+            tokens[token.address] = token
+
+        return cls(tokens)
+
+    def get_token(self, address: bytes) -> SolanaTokenInfo:
+        return self._tokens.get(
+            address,
+            SolanaTokenInfo(
+                address=b"",
+                # REVIEW: empty?
+                program_id="",
+                ticker="[TOKEN]",
+            ),
```

Comment:
Let's extract this into a `UNKNOWN_TOKEN` variable as is done in `ethereum/definitions.py` (but we can keep the var in this for now I guess) 🙏